src/shared/pkcs11-util.c

```diff
@@ -523,6 +523,272 @@ int pkcs11_token_find_x509_certificate(
 }
 
 #if HAVE_OPENSSL
+int pkcs11_token_read_public_key(
+                CK_FUNCTION_LIST *m,
+                CK_SESSION_HANDLE session,
+                CK_OBJECT_HANDLE object,
+                EVP_PKEY **ret_pkey) {
+
+        CK_KEY_TYPE key_type;
+        CK_ATTRIBUTE attributes[] = {
+                { CKA_PUBLIC_KEY_INFO, NULL_PTR,  0                },
+                { CKA_KEY_TYPE,        &key_type, sizeof(key_type) }
```

Comment:
we make it a rule to always put a "," after the last element in arrays like this which span multiple lines, to keep diffs small